test/common/config/subscription_factory_impl_test.cc

```diff
@@ -328,6 +331,48 @@ TEST_F(SubscriptionFactoryTest, GrpcSubscription) {
   subscriptionFromConfigSource(config)->start({"static_cluster"});
 }
 
+TEST_F(SubscriptionFactoryTest, GrpcCollectionSubscriptionBadType) {
+  EXPECT_THROW_WITH_MESSAGE(collectionSubscriptionFromUrl("xdstp:///foo", {})->start({}),
+                            EnvoyException,
+                            "xdstp:// type does not match "
+                            "envoy.config.endpoint.v3.ClusterLoadAssignment in xdstp:///foo");
+}
+
+TEST_F(SubscriptionFactoryTest, GrpcCollectionSubscriptionUnsupportedApiType) {
+  envoy::config::core::v3::ConfigSource config;
+  auto* api_config_source = config.mutable_api_config_source();
+  api_config_source->set_api_type(envoy::config::core::v3::ApiConfigSource::DELTA_GRPC);
+  api_config_source->set_transport_api_version(envoy::config::core::v3::V3);
+  api_config_source->add_grpc_services()->mutable_envoy_grpc()->set_cluster_name("static_cluster");
+  Upstream::ClusterManager::ClusterSet primary_clusters;
+  primary_clusters.insert("static_cluster");
+  EXPECT_CALL(cm_, primaryClusters()).WillOnce(ReturnRef(primary_clusters));
+  EXPECT_THROW_WITH_REGEX(
+      collectionSubscriptionFromUrl(
+          "xdstp://foo/envoy.config.endpoint.v3.ClusterLoadAssignment/bar", config)
+          ->start({}),
+      EnvoyException, "Unknown xdstp:// transport API type in api_type: DELTA_GRPC");
```

Comment:
Why doesn't this work?  Does this mean that new-style resource names require the aggregated incremental protocol variant specifically?  It's not clear to me why we require the aggregated variant here -- it seems like this should also work for non-aggregated incremental.

Reply:
Limitation of implementation. The idea is not to boil the ocean in this PR but add some basic support, we can add non-ADS in the next PR. I can clarify in the commit message.